akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -59,38 +59,81 @@ private[akka] final case class TakeWhile[T](p: T ⇒ Boolean, decider: Supervisi
 /**
  * INTERNAL API
  */
-private[akka] final case class DropWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
-  var taking = false
+private[stream] final case class DropWhile[T](p: T ⇒ Boolean) extends GraphStage[FlowShape[T, T]] {
+  val in = Inlet[T]("DropWhile.in")
+  val out = Outlet[T]("DropWhile.out")
+  override val shape = FlowShape(in, out)
+  override def initialAttributes: Attributes = DefaultAttributes.dropWhile
 
-  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
-    if (taking || !p(elem)) {
-      taking = true
-      ctx.push(elem)
-    } else {
-      ctx.pull()
+  def createLogic(inheritedAttributes: Attributes) = new SupervisedGraphStageLogic(inheritedAttributes, shape) with InHandler with OutHandler {
+    override def onPush(): Unit = execute(() ⇒ {
+      val elem = grab(in)
+      if (p(elem)) pull(in)
+      else {
+        push(out, elem)
+        setHandler(in, rest)
+      }
+    })
```

Comment:
should I also add what is suggested in https://github.com/akka/akka/pull/19990/files/e0d4b2e4ecf1a0559ccb477e690b6350368f5eec#r55859774 ?